gcloud/bigquery/table.py

```diff
@@ -0,0 +1,237 @@
+        """
+        return self._properties.get('type')
+
+    @property
+    def description(self):
+        """Description of the table.
+
+        :rtype: string, or ``NoneType``
+        :returns: The description as set by the user, or None (the default).
+        """
+        return self._properties.get('description')
+
+    @description.setter
+    def description(self, value):
+        """Update description of the table.
+
+        :type value: string, or ``NoneType``
+        :param value: new description
+
+        :raises: ValueError for invalid value types.
+        """
+        if not isinstance(value, six.string_types) and value is not None:
+            raise ValueError("Pass a string, or None")
+        self._properties['description'] = value
+
+    @property
+    def expires(self):
+        """Datetime at which the table will be removed.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        :returns: the expiration time, or None
+        """
+        return _datetime_from_prop(self._properties.get('expirationTime'))
+
+    @expires.setter
+    def expires(self, value):
+        """Update atetime at which the table will be removed.
+
+        :type value: ``datetime.datetime``, or ``NoneType``
+        :param value: the new expiration time, or None
+        """
+        if not isinstance(value, datetime.datetime) and value is not None:
+            raise ValueError("Pass a datetime, or None")
+        self._properties['expirationTime'] = _prop_from_datetime(value)
+
+    @property
+    def friendly_name(self):
+        """Title of the table.
+
+        :rtype: string, or ``NoneType``
+        :returns: The name as set by the user, or None (the default).
+        """
+        return self._properties.get('friendlyName')
+
+    @friendly_name.setter
+    def friendly_name(self, value):
+        """Update title of the table.
+
+        :type value: string, or ``NoneType``
+        :param value: new title
+
+        :raises: ValueError for invalid value types.
+        """
+        if not isinstance(value, six.string_types) and value is not None:
+            raise ValueError("Pass a string, or None")
+        self._properties['friendlyName'] = value
+
+    @property
+    def location(self):
+        """Location in which the table is hosted.
+
+        :rtype: string, or ``NoneType``
+        :returns: The location as set by the user, or None (the default).
+        """
+        return self._properties.get('location')
+
+    @location.setter
+    def location(self, value):
+        """Update location in which the table is hosted.
+
+        :type value: string, or ``NoneType``
+        :param value: new location
+
+        :raises: ValueError for invalid value types.
+        """
+        if not isinstance(value, six.string_types) and value is not None:
+            raise ValueError("Pass a string, or None")
+        self._properties['location'] = value
+
+    @property
+    def view_query(self):
```

Comment:
Typical for trying to keep relational terminology around, though.  It is actually crappier wedging views in on top of tables by my lights.